basic-services/TraditionalClient.cs

```diff
@@ -1,81 +1,292 @@
 ﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
 using System.Collections.Generic;
 using System.Globalization;
+using Flurl;
+using Flurl.Http;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Threading;
 
 namespace JohnsonControls.Metasys.BasicServices
 {
     public class TraditionalClient
     {
+        private FlurlClient client;
+
+        private string accessToken;
+
+        private DateTime tokenExpires;
+
+        private bool refresh;
+
+        private const int MAX_PAGE_SIZE = 1000;
 
         /// <summary>
-        /// 
+        /// Creates a new TraditionalClient.
         /// </summary>
         /// <remarks>
         /// Takes an optional CultureInfo which is useful for formatting numbers. If not specified,
         /// the user's current culture is used.
         /// </remarks>
         /// <param name="cultureInfo"></param>
-        /// <exception cref="NotImplementedException"></exception>
         public TraditionalClient(CultureInfo cultureInfo = null)
         {
             var culture = cultureInfo ?? CultureInfo.CurrentCulture;
-            throw new NotImplementedException();
+            FlurlHttp.Configure(settings => settings.OnErrorAsync = HandleFlurlErrorAsync);
+            FlurlHttp.Configure(settings => settings.OnError = HandleFlurlError);
+        }
+
+        private void HandleFlurlError(HttpCall call)
+        {
+            var task = HandleFlurlErrorAsync(call);
+        }
+
+        private async Task HandleFlurlErrorAsync(HttpCall call)
+        {
+            if (call.Exception.GetType() != typeof(Flurl.Http.FlurlParsingException))
+            {
+                await LogErrorAsync(call.Exception.Message);
+            }
+            call.ExceptionHandled = true;
+        }
+
+        private async Task LogErrorAsync(String message)
+        {
+            await Console.Error.WriteLineAsync(message);
+        }
+
+        private void LogClientUndefinedError()
+        {
+            var task = LogErrorAsync("Client undefined please login with TryLogin method.");
         }
 
-        // Probably a boolean isn't good enough as it doesn't give a client any indication of what went wrong.
-        // Could return an Result structure that contains error information or nothing.
-        public bool TryLogin(string username, string password, string hostname, ApiVersion version = ApiVersion.V2)
+        /// <summary>
+        /// Attempts to login to the given host.
+        /// </summary>
+        /// <exception cref="Flurl.Http.FlurlHttpException"></exception>
+        /// <exception cref="System.NullReferenceException"></exception>
+        public void TryLogin(string username, string password, string hostname, ApiVersion version = ApiVersion.V2, bool refresh = true)
+        {
+            this.refresh = true;
+            if (client != null)
```

Comment:
I would not allow multiple logins. 

Instead I would enforce the invariant that `client` is never null by creating it in the constructor. If that means moving the `hostname` into the constructor that's fine. Then you can remove lots of client null checks.

Reply:
I read that for COM the constructor should not accept any parameters. If we make this change it would affect how the LegacyClient is implemented. I see the null checks being necessary in the LegacyClient if we add the parameter to the TraditionalClient constructor.

Reply:
> I read that for COM the constructor should not accept any parameters

That's correct.

> If we make this change it would affect how the LegacyClient is implemented. 

That's okay. In that case we're designing to the constraints of the framework (COM). Or we can create a ClientFactory that takes no parameters but has a CreateClient method that takes a hostname. In that way, we can also design the LegacyClient without all the null checks. See Factory Pattern

Reply:
I added the hostname and version to the constructor.